mdata/store_mock.go

```diff
@@ -0,0 +1,48 @@
+package mdata
+
+import "github.com/raintank/metrictank/mdata/chunk"
+
+type mockSearchResult struct {
+	chunks []chunk.IterGen
+	err    error
+}
+
+type mockStore struct {
+	CurrCall int
+	Results  []mockSearchResult
+}
+
+func NewMockStore() *mockStore {
+	d := &mockStore{
+		CurrCall: 0,
+		Results:  make([]mockSearchResult, 0),
+	}
+	return d
+}
+
+func (c *mockStore) AddMockResult(chunks []chunk.IterGen, err error) {
+	// copy chunks because we don't want to modify the source
```

Comment:
does it happen that things get modified after a Search? what exactly (the itergen properties? the data in the slice?) maybe we should add a comment not to do that.  
also I believe this will just copy the itergens, changes made to an IterGen.b (or rather: within the array that the slice points to) will be visible in other places, I think.